components/config-mgmt-service/cmd/chef-automate-collect/commands/gen-config.go

```diff
@@ -0,0 +1,107 @@
+package commands
+
+import (
+	"os"
+
+	"github.com/BurntSushi/toml"
+	"github.com/spf13/cobra"
+)
+
+var genConfigCommands = struct {
+	verbose            bool
+	insecureConnection bool
+	privateConfig      bool
+	writeRepoConfig    bool
+	writeUserConfig    bool
+}{}
+
+func newGenConfigCommand() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "gen-config AUTOMATE_URL AUTOMATE_AUTH_TOKEN",
+		Short: "verify config parameters and emit config",
+		Long: `
+		gen-config makes a request to Automate to verify the Automate URL and
+		token. If successful, it emits 
+		`,
+		Args: cobra.ExactArgs(2),
+		RunE: runGenConfigCommand,
+	}
+	c.Flags().BoolVarP(&genConfigCommands.verbose,
+		"verbose",
+		"v",
+		false,
+		"enable verbose output (goes to stderr)",
+	)
+	c.Flags().BoolVarP(&genConfigCommands.insecureConnection,
+		"insecure",
+		"k",
+		false,
+		"do not verify the Automate host's TLS certificates/hostname",
+	)
+	c.Flags().BoolVarP(&genConfigCommands.privateConfig,
+		"private",
+		"p",
+		false,
+		"emit private config with credentials",
+	)
+	c.Flags().BoolVar(&genConfigCommands.writeRepoConfig,
+		"repo",
+		false,
+		"write config suitable for use in a shared SCM repo",
+	)
+	c.Flags().BoolVar(&genConfigCommands.writeUserConfig,
+		"user",
+		false,
+		"write config for your user account",
+	)
+	return c
+}
+
+func runGenConfigCommand(c *cobra.Command, args []string) error {
+	automateURLIn := args[0]
+	token := args[1]
+
+	cliIO.EnableVerbose = genConfigCommands.verbose
+
+	ac, err := newAutomateConfig(automateURLIn, token)
+	if err != nil {
+		return err
+	}
+	ac.InsecureTLS = genConfigCommands.insecureConnection
+	conf := &Config{Automate: ac}
+
+	err = ac.Test()
+	if err != nil {
+		return err
+	}
+
+	cliIO.msg("# Config verified by successful request to %q", ac.TestURL)
```

Comment:
This line does not seem to work with %q. I got the message "Config verified by successful request to %!q(func() (*url.URL, error)=0x7d4360)"

Reply:
oops, that's a bug. will fix 👍